### PR TITLE
ci: support sles for hal cluster

### PR DESCRIPTION
**Pull request by yangchiu**
#### Which issue(s) this PR fixes:

Issue https://github.com/longhorn/longhorn/issues/9691

#### What this PR does / why we need it:

support sles for hal cluster

## Summary by CodeRabbit

- **New Features**
	- Introduced a Terraform configuration for managing a Rancher2 environment with Harvester integration.
	- Added support for cloud credential management and machine configuration for Kubernetes deployment.
	- New variables for enhanced configuration management, including sensitive access keys and Kubernetes distribution settings.
	- Enhanced logic for determining registration codes based on distribution and architecture parameters.

- **Bug Fixes**
	- Improved resource naming and configuration retrieval for existing clusters.

### CODE REVIEW

File: test_framework/terraform/harvester/sles/main.tf [hunk 1]
```diff
@@ -0,0 +1,208 @@
+terraform {
+  required_providers {
+    rancher2 = {
+      source  = "rancher/rancher2"
+      version = "~> 4.4.0"
```
Comment (khushboo-rancher) [inline test_framework/terraform/harvester/sles/main.tf:5]: Can we pivot to [v5.1.0](https://github.com/rancher/terraform-provider-rancher2/releases/tag/v5.1.0) latest version? v5.x has some fixes.

**Review by khushboo-rancher — COMMENTED** [review]
LGTM, couple of review comments from coderabbit looks ok to address.

File: test_framework/terraform/harvester/sles/variables.tf [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+variable "k8s_distro_name" {
+  type        = string
+  default     = "rke2"
+  description = "kubernetes distro version to install [rke, k3s]  (default: rke2)"
+}
+
+variable "k8s_distro_version" {
+  type        = string
+  default     = "v1.27.13+rke2r1"
```
Comment (roger-ryao) [inline test_framework/terraform/harvester/sles/variables.tf:43]: The HAL server is currently at version `v1.28.14+rke2r1` by default.
```suggestion
  default     = "v1.28.14+rke2r1"
```

File: test_framework/terraform/harvester/sles/variables.tf [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  description = <<-EOT
+    kubernetes version that will be deployed
+    k3s: (default: v1.27.13+k3s1)
```
Comment (roger-ryao) [inline test_framework/terraform/harvester/sles/variables.tf:46]:
```suggestion
    k3s: (default: v1.28.14+k3s1)
```

File: test_framework/terraform/harvester/sles/variables.tf [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    rke2: (default: v1.27.13+rke2r1)
```
Comment (roger-ryao) [inline test_framework/terraform/harvester/sles/variables.tf:47]:
```suggestion
    rke2: (default: v1.28.14+rke2r1)
```

File: test_framework/terraform/harvester/sles/variables.tf [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+variable "os_distro_version" {
+  type        = string
+  default     = "15-sp6"
+}
+
+variable "k8s_distro_name" {
+  type        = string
+  default     = "rke2"
+  description = "kubernetes distro version to install [rke, k3s]  (default: rke2)"
```
Comment (roger-ryao) [inline test_framework/terraform/harvester/sles/variables.tf:38]: `rke` should be a typo.
```suggestion
  description = "kubernetes distro version to install [rke2, k3s]  (default: rke2)"
```

File: test_framework/terraform/harvester/sles/main.tf [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+resource "rancher2_machine_config_v2" "e2e-machine-config-worker" {
+
+  generate_name = "e2e-machine-config-worker-${random_string.random_suffix.id}"
+
+  harvester_config {
+
+    vm_namespace = "longhorn-qa"
+
+    cpu_count = "4"
+    memory_size = "8"
```
Comment (roger-ryao) [inline test_framework/terraform/harvester/sles/main.tf:93]:
```suggestion
    memory_size = "16"
```
I recall @derekbit saying that if we want to use a `v2 volume`, it seems that `16 GB` of host memory is more appropriate.

Comment (yangchiu) [reply test_framework/terraform/harvester/sles/main.tf:93]: @derekbit Where can we find documentation on this?

File: test_framework/terraform/harvester/sles/main.tf (hunk 1 above)
Comment (yangchiu) [reply test_framework/terraform/harvester/sles/main.tf]: Refined.

File: test_framework/terraform/harvester/sles/main.tf (hunk 6 above)
Comment (derekbit) [reply test_framework/terraform/harvester/sles/main.tf:93]: 8 GiB should be okay. Can we try it?

Comment (yangchiu) [reply test_framework/terraform/harvester/sles/main.tf:93]: > 8 GiB should be okay. Can we try it?

It's been tested.

**Review by roger-ryao — APPROVED** [review]
LGTM

**Pull request merged**